lib/home_page.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:pokedex/features/pokemon_favorites/favorite_connector.dart';
+import 'package:pokedex/features/pokemon_overview/pokemon_list_connector.dart';
+import 'package:pokedex/utils/strings.dart' as str;
+
+class HomePage extends StatelessWidget {
+  const HomePage({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    final router = GoRouter.of(context);
+    final currentNavIndex = _selectNavIndex(router.location);
+    final navBarItems = <BottomNavigationBarItem>[
+      const BottomNavigationBarItem(
+        label: str.homeTitle,
+        icon: Icon(Icons.home),
+      ),
+      const BottomNavigationBarItem(
+        label: str.favoriteTitle,
+        icon: Icon(Icons.favorite),
+      ),
+    ];
+
+    return Scaffold(
+      body: child,
+      bottomNavigationBar: BottomNavigationBar(
+        backgroundColor: Colors.blue,
+        selectedItemColor: Colors.white,
+        unselectedItemColor: Colors.white38,
+        iconSize: 25,
+        currentIndex: currentNavIndex,
+        showUnselectedLabels: false,
+        onTap: (index) => context.go(_onNavigateToTab(index, currentNavIndex)),
+        items: navBarItems,
+      ),
+    );
+  }
+
+  int _selectNavIndex(String routerLocation) {
+    switch (routerLocation) {
+      case PokemonListConnector.route:
+        return 0;
+      case FavoriteConnector.route:
+        return 1;
+      default:
+        return 0;
+    }
+  }
+
+  String _selectNavPath(int selectedTabIndex) {
+    switch (selectedTabIndex) {
+      case 0:
+        return PokemonListConnector.route;
+      case 1:
+        return FavoriteConnector.route;
+      default:
+        return PokemonListConnector.route;
+    }
+  }
+
+  String _onNavigateToTab(int selectedTabIndex, int currentTabIndex) {
```

Comment:
Prefix "on" is most often used to indicate that method is intended to be used as a callback, i.e. not called directly, but set as a handler for some event.

For example, when you write method named onClick, you probably won't call it directly, but rather expect that GUI toolkit will call it once user clicks a button.

Now the thing here is that you have used the prefix `"on"` and yet you're enforcing this function to supply a return object of String? 🙈 

instead of misusing the naming you gave for this function, how about if you go for this solution and have this passed immediately on BottomNavigationBar's `onTap` parameter 🙂 :

```
void _onNavigateToTab(int selectedTabIndex, int currentTabIndex) {
    if (currentTabIndex == selectedTabIndex) return;

    final navPath = _selectNavPath(selectedTabIndex);
    context.go(navPath);
  }
```

Reply:
the problem with that function is that there is no context present outside the widget build function

Reply:
that's taken from my previous example when this widget was still stateful, for this instance can't you add a context param on that function if that's the case? 🙂 
also don't forget to apply trailing commas 😜